server/channels/app/limits_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"testing"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetUserLimits(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("base case", func(t *testing.T) {
+		userLimits, appErr := th.App.GetUserLimits()
+		require.Nil(t, appErr)
+
+		// InitBasic creates 3 users by default
+		require.Equal(t, int64(3), userLimits.ActiveUserCount)
+		require.Equal(t, int64(10000), userLimits.MaxUsersLimit)
+		require.Equal(t, int64(500), userLimits.LowerBandUserLimit)
+		require.Equal(t, int64(9000), userLimits.UpperBandUserLimit)
+	})
+
+	t.Run("user count should increase on creating new user and decrease on deleting", func(t *testing.T) {
+		userLimits, appErr := th.App.GetUserLimits()
+		require.Nil(t, appErr)
+		require.Equal(t, int64(3), userLimits.ActiveUserCount)
+
+		// now we create a new user
+		newUser := th.CreateUser()
+
+		userLimits, appErr = th.App.GetUserLimits()
+		require.Nil(t, appErr)
+		require.Equal(t, int64(4), userLimits.ActiveUserCount)
+
+		// now we'll delete the user
+		_ = th.App.PermanentDeleteUser(th.Context, newUser)
```

Comment:
It is nice to test with permanent deletion, but the normal way to "delete users" is just deactivating them. We should test for that too.

Reply:
Good catch! Done, added,